server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -1025,7 +1025,14 @@ protected String delegatingTo() {
             }
             SourceValueFetcher fetcher = SourceValueFetcher.toString(blContext.sourcePaths(name()));
             var sourceMode = blContext.indexSettings().getIndexMappingSourceMode();
-            return new BlockSourceReader.BytesRefsBlockLoader(fetcher, blockReaderDisiLookup(blContext), sourceMode);
+            if (sourceMode == SourceFieldMapper.Mode.SYNTHETIC && syntheticSourceDelegate().ignoreAbove() != Integer.MAX_VALUE) {
+                // Can't just read the original field, because only when ignore_above threshold is exceeded,
+                // then this stored fields gets added. In other cases this field does not exist. But we do need to include the field name:
+                String originalName = syntheticSourceDelegate().name() + "._original";
```

Comment:
Can we encapsulate this logic inside `KeywordFieldMapper` somehow similar to how `syntheticSourceDelegate()` works? It would eliminate this kind of problem.